### PR TITLE
Add sok_tr spider (~11000 locations across Turkey)

### PR DESCRIPTION
**Pull request by styrowolf**
ŞOK is one of the largest Turkish supermarket chains with around 11000 locations in Turkey.

Wikidata: https://www.wikidata.org/wiki/Q19613992

### CODE REVIEW

**Comment by CloCkWeRX** [comment]
Ah nice, I feel I got blocked by Cloudflare when I looked at this last.

### Brand name

Şok Supermarket Chains

Turkish supermarket

### Wikidata ID

Q19613992
https://www.wikidata.org/wiki/Q19613992
https://www.wikidata.org/wiki/Special:EntityData/Q19613992.json

### Store finder url(s)

Official Url(s): https://www.sokmarket.com.tr/
- [x] I have run the below and if anything in generated, added it.
- [ ] I have manually checked for a storefinder
- [ ] This is not a manufacturer listing outlets operating under their own brands / this will not create duplicates
`pipenv run scrapy sf --brand-wikidata=Q19613992 https://www.sokmarket.com.tr/`
```
```


Cloudflare

File: locations/spiders/sok_tr.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+from locations.pipelines.address_clean_up import clean_address, merge_address_lines
+
+PROVINCES_URL = "https://kurumsal.sokmarket.com.tr/ajax/servis/sehirler"
+DISTRICTS_URL = "https://kurumsal.sokmarket.com.tr/ajax/servis/ilceler?city={province}"
+STORES_URL = "https://kurumsal.sokmarket.com.tr/ajax/servis/magazalarimiz?city={province}&district={district}"
+
+
+class SokTRSpider(scrapy.Spider):
+    name = "sok_tr"
+    item_attributes = {"brand": "ŞOK", "brand_wikidata": "Q19613992"}
```
Comment (Cj-Malone) [inline locations/spiders/sok_tr.py:17]: NSI has this as `Şok`

https://nsi.guide/index.html?t=brands&k=shop&v=supermarket&tt=q19613992

Comment (styrowolf) [reply locations/spiders/sok_tr.py:17]: I looked at the website and the naming is not consistent. They sometimes use the name with all caps when referring to the chain stores ("ŞOK marketleri") or sometimes with normal capitalization. I can change it to "Şok" if it is more appropriate.

Pages I've looked at:
https://kurumsal.sokmarket.com.tr/tarihcemiz
https://kurumsal.sokmarket.com.tr/insan-kaynaklari/insan-kaynaklari-politikamiz
https://kurumsal.sokmarket.com.tr/markalar/soka-ozel-urunler

Comment (Cj-Malone) [reply locations/spiders/sok_tr.py:17]: I have a strong preference being the same as NSI. But if NSI is wrong, it can be changed there.

**Pull request merged**